homeassistant/components/smlight/button.py

```diff
@@ -0,0 +1,86 @@
+"""Support for SLZB-06 buttons."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+import logging
+from typing import Final
+
+from pysmlight.web import CmdWrapper
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import SmDataUpdateCoordinator
+from .entity import SmEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class SmButtonDescription(ButtonEntityDescription):
+    """Class to describe a Button entity."""
+
+    entity_category = EntityCategory.CONFIG
+    press_fn: Callable[[CmdWrapper], Awaitable[None]]
+
+
+BUTTONS: Final = [
+    SmButtonDescription(
+        key="core_restart",
+        translation_key="core_restart",
+        device_class=ButtonDeviceClass.RESTART,
+        press_fn=lambda cmd: cmd.reboot(),
+    ),
+    SmButtonDescription(
+        key="zigbee_restart",
+        translation_key="zigbee_restart",
+        device_class=ButtonDeviceClass.RESTART,
+        press_fn=lambda cmd: cmd.zb_restart(),
+    ),
+    SmButtonDescription(
+        key="zigbee_flash_mode",
+        translation_key="zigbee_flash_mode",
+        press_fn=lambda cmd: cmd.zb_bootloader(),
+    ),
```

Comment:
what does this do? sounds dangerous, should we disable it by default so only people who need it can enable it?

Reply:
It's not really dangerous, just reboots into a mode waiting for flash. It goes back to normal after a subsequent restart of Zigbee chip.

However will disable it by default, since most users should just use the update sensors for firmware updates (once they are merged in later PR)